common/test_local_filestore.py

```diff
@@ -59,6 +59,19 @@ def test_cp(tmp_path):
         assert file_handle.read() == data
 
 
+def test_non_exist_dest(tmp_path):
+    """Tests cp and rsync will create intermediate folders for destination."""
+    source_dir = tmp_path / 'source'
+    source_dir.mkdir()
+    source_file = source_dir / 'file1'
+    open(source_file, "w+").close()
+
+    cp_dest_dir = tmp_path / 'cp_test' / 'intermediate' / 'cp_dest'
+    rsync_dest_dir = tmp_path / 'rsync_test' / 'intermediate' / 'rsync_dest'
+    local_filestore.cp(str(source_dir), str(cp_dest_dir), recursive=True)
```

Comment:
What is this for?

Reply:
We test `cp` and `rsync` together here.

Reply:
Ah can you break this up into two test please?

Reply:
OK. Done.